### ISSUE TITLE
Hold Input Button Bug

### ISSUE THREAD

**Issue by wesleywh** [wesleywh/EasyMultiplayer-Invector#27]
**Describe the bug**
Reported by @lorax
```
found an odd bug. Don't think it is specific to my character controller.
When holding a weapon (vAxe), if i click and hold my client will play the attack animation once.
Other clients will see me continue to attack as long as the button is held.
```

**To Reproduce**
 "Hosted Mode"
1. Run ShooterInventory scene
2.Pickup Axe
3. Hold Attack button
4. Owner client attacks once, everyone else continues to attack

**Expected behavior**
Attacks should be synced with their number of attacks.

**Screenshots/YouTube Links**

https://www.loom.com/share/923636e20589447ebc5bf3aba077a07d

**Versions**
 - OS: Windows
 - Unity Version 2020.3
 - Invector Version 2.6.2c
 - EMI Package - FullSuite
 - EMI Version 0.5.1

**Comment by wesleywh** [comment]
Confirmed.

**Comment by wesleywh** [comment]
This is happening because a switch happens when you equip a melee weapon inside the 
```
protected virtual bool DrawWeaponsImmediateConditions()
```
function that lives in the `vDrawHideMeleeWeapons` component. I'm suspecting that the `melee.weakAttackInput.GetButton()` is triggering a reset to be sent to the server every second when the button is held. Not a 100% confirmed yet but I bet the issue lives in the `MP_vInput` component in the `GetButton` function.

**Comment by wesleywh** [comment]
I figured out what is going on. It's going to take a little bit to implement a fix. Basically it is this.

The `curBoolValue` and `prevBoolValue` are shared between all methods `GetButtonDown`, `GetButton`, `GetButtonUp`, etc. For certain button presses this is okay but for others, like the `WeakAttackInput` this MUST be separated out to individual actions. So I made the following new variables:
```
curHoldBoolValue
curUpBoolValue
curDownBoolValue
prevHoldBoolValue
prevUpBoolValue
prevDownBoolValue
```
In the `MP_vInput` component. I also added some additional logic to check for button names with keywords to make sure that `down` and `up` buttons are reset properly as with the new setup it will only send to the server on a `true` result. So the server will need to reset the buttons to a `false` result to properly work. There is a lot to work through but that's the current issue. I will add a hotfix to the `#downloads` channel when I have completed my work.
